toripchanger/changer.py

```diff
@@ -186,7 +187,7 @@ def _obtain_new_ip(self):
         Change Tor's IP.
         """
         with Controller.from_port(
-            address=self.tor_address, port=self.tor_port
+            address=socket.gethostbyname(self.tor_address), port=self.tor_port
```

Comment:
Please don't do this. The method purpose is to obtain a new Tor IP, nothing else.
You can always do something like `TorIpChanger(tor_address=socket.gethostbyname("localhost"))`, but I don't think `_obtain_new_ip()` should be responsible for doing this.

Reply:
Okay.
Don't you want the package to do name resolution at all (i.e somewhere else in the code) ? That would remove a burden on the user to do the name resolution (him/her)self...

Reply:
Well, to be honest I don't know :man_shrugging:. I am not saying no. `Controller.from_port` expects an IP as the value of `address`, that's why `TorIpChanger` follows. I consider `TorIpChanger` as a thin wrapper around Stem. 

What's the use-case of using a host name instead of an IP, please?

Reply:
To elaborate a bit more. I think it's too late to do the host name resolution in `_obtain_new_ip()`. What if it fails? [To be fair there's no validation for `tor_address` either.] I would rather do it in the constructor, but that would mean a breaking change to the API. One way out of this is to add a second class, based on `TorIpChanger`, and override its constructor. But what is a sensible name such class?

Reply:
The thing is that humans tend to prefer calling a server by its hostname, and not its ip. In my case, I have a docker stack with tor in a container of its own, and my scraper runs from inside a scrapyd container. I deploy the whole thing with docker compose, thus, locally, each of the 2 container can make a request to the other using its container name (used as its hostname).
I think we can agree on resolving the `tor_address` at initialisation of `TorIpChanger` class, validating the given tor address at the same time. So by the time we arrive at `_obtain_new_ip()`, we are already sure it will not fail because of bad hostname or ip.

The other config parameter (`local_http_proxy`) already exposes the versatility of accepting hostname or ip address (because it gets passed to `requests` as is, and `requests` offers that flexibility). I think allowing `TorIpChanger` to accept hostnames would preserve consistency in that respect.

Reply:
OK, fair enough. I'll try to look into it a bit more during the weekend. Feel free to open another PR if you have it implemented already.